internal/files/files.go

```diff
@@ -1,45 +1,32 @@
 package files
 
 import (
-	"fmt"
 	"sort"
-	"strings"
 
-	"github.com/goreleaser/nfpm"
 	"github.com/goreleaser/nfpm/glob"
 )
 
-// FileToCopy describes the source and destination of one file to copy into a
-// package and whether it is a config file.
+// FileToCopy describes the source and destination
+// of one file to copy into a package.
 type FileToCopy struct {
 	Source      string
 	Destination string
-	Config      bool
 }
 
-// ToCopy lists all of the real files to be copied into the package.
-func ToCopy(info *nfpm.Info) ([]FileToCopy, error) {
+// Expand gathers all of the real files to be copied into the package.
+func Expand(filesSrcDstMap map[string]string) ([]FileToCopy, error) {
 	var files []FileToCopy
-	for i, filesMap := range []map[string]string{info.Files, info.ConfigFiles} {
-		for srcglob, dstroot := range filesMap {
-			globbed, err := glob.Glob(srcglob, dstroot)
-			if err != nil {
-				return nil, err
-			}
-			for src, dst := range globbed {
-				// avoid including a partial file with the name of the target in the target
-				// itself. when used as a lib, target may not be set. in that case, src will
-				// always have the empty sufix, and all files will be ignored.
-				// TODO: add tests cases for this
-				if info.Target != "" && strings.HasSuffix(src, info.Target) {
```

Comment:
As a changed the signature I had to think about if I still want to pass in the target to do this check. But then I thought we should remove this check because, in my opinion, it has no business being here. If I want to include a file that ends with "deb" or "rpm" in my package I should be able to. If I accidentally glob files I do not want, that's a user error which should not be corrected magically by some heuristic in `nfpm`.

Some examples: I could have a file named `harddrives_rpm` containing rounds per minute specs of my hard drives. Or apparently `ardeb` is a unit of dry measure used in egypt (wow, this is getting a bit far fetched, sorry 😄). Both files could never end up in an package.

Reply:
makes sense to me :)